app/controllers/settings/dex_connector_oidcs_controller.rb

```diff
@@ -0,0 +1,88 @@
+# Settings::DexConnectorOidcsController is responsible to manage requests
+# related to OIDC connectors.
+class Settings::DexConnectorOidcsController < SettingsController
+  # populate @data_holder with existing database record
+  before_action :set_data_holder, except: [:index, :new, :create]
+
+  def index
+    @oidc_connectors = DexConnectorOidc.all
+  end
+
+  def new
+    @is_data_valid = false
+    @data_holder = data_holder_type.new
+  end
+
+  def edit
+    @is_data_valid = false
+  end
+
+  def create
+    @data_holder = data_holder_type.new(data_holder_params)
+
+    if params[:validate]
+      @is_data_valid = @data_holder.valid?
+      # TODO: figure out how to make notice actually display :)
```

Comment:
i think they should be displayed/rendered via the layout, see https://github.com/kubic-project/velum/blob/master/app/views/layouts/settings.html.slim#L28 - @vitoravelino any idea here?

Reply:
I assume it's a difference between `redirect_to` v/s `render`.  This is my first time working with Rails, so it wasn't obvious, and then I forgot to look that up after I got the rest working. :)

I did a little research this morning, now that I was reminded of this TODO.  It appears that render doesn't take the flash as input?  Either way, if I directly assign to `flash[:notice]` or `flash[:alert]` before the render, that works properly with the `render`.  I'm not positive if that's the "right" way to do it, though.  How about this, @MaximilianMeister @vitoravelino?
```ruby
      if @is_data_valid
        flash.now[:notice] = "#{@data_holder.class} is valid"
      else
        flash.now[:alert]  = "#{@data_holder.class} is invalid"
      end
      render action: :edit
```

I needed to use flash.now so the "is invalid" message doesn't stay visible when it's corrected (or vice-versa).

Reply:
Well, `render action: :new` here; I tested it on the edit page. :)

Reply:
There - works properly in f0fbdcf.